src/Frame/VectorFieldCalculator.cc

```diff
@@ -0,0 +1,296 @@
+/* This file is part of the CARTA Image Viewer: https://github.com/CARTAvis/carta-backend
+   Copyright 2018, 2019, 2020, 2021 Academia Sinica Institute of Astronomy and Astrophysics (ASIAA),
+   Associated Universities, Inc. (AUI) and the Inter-University Institute for Data Intensive Astronomy (IDIA)
+   SPDX-License-Identifier: GPL-3.0-or-later
+*/
+
+#include "VectorFieldCalculator.h"
+
+#include "DataStream/Compression.h"
+
+namespace carta {
+
+bool VectorFieldCalculator::DoCalculations(VectorFieldCallback& callback) {
+    if (!frame) {
```

Comment:
Is this enough? I assume that before the vector field calculator was running in a valid frame, should you check `frame->IsValid()`?.